### DISCUSSION TITLE
Is it possible to use unified to generate multiple outputs?

### DISCUSSION THREAD

**Discussion by xiaoxinghu** [unifiedjs/unified#103] [Q&A] [answered]
Use case: instead of having a 1 to 1 relationship between a markdown file and a `hast`, I want each level 1 heading to become a new input. And still, be able to apply further plugins/transpilers to each of them after that plugin.

The input file:
```md
# my first post
...
# my second post
...
# another post
...
```

The processor:
```js
unified()
  .use(remarkParse)
  .use(splitByHeading) // after this point, we have multiple hast to be processed by the following plugins.
  .use(remark2rehype)
  .use(toc)
```

At the end of the processing, we have an array.

**Comment by wooorm** [comment, marked as answer]
The process deals with one tree and file indeed. But with https://github.com/unifiedjs/unified/discussions/104, you’ll end up with a parse/run pipeline for the whole file, and run/stringify pipelines for the separate files.

**Reply by xiaoxinghu** [reply]
Yeah, that will do. Thanks for the super faster reply, as always.
